File: ios-swiftui/library/Sources/PlanetComponents/TspChip.swift
import SwiftUI

public struct TspChip: View {
    let text: String
    let selected: Bool
    let disabled: Bool
    let theme: StarPlanetTheme
    let action: () -> Void

    public init(_ text: String, selected: Bool = false, disabled: Bool = false, theme: StarPlanetTheme = .sky, action: @escaping () -> Void = {}) {
        self.text = text
        self.selected = selected
        self.disabled = disabled
        self.theme = theme
        self.action = action
    }

    public var body: some View {
        Button(action: { if !disabled { action() } }) {
            Text(text).font(.system(size: 14, weight: .bold)).foregroundColor(theme.textPrimary)
                .padding(.horizontal, 14).padding(.vertical, 8)
                .background(selected ? theme.selectedFill : theme.surfaceRaised)
                .overlay(Capsule().stroke(selected ? theme.success : theme.borderDefault, lineWidth: 1))
                .clipShape(Capsule())
        }
        .buttonStyle(.plain)
        .opacity(disabled ? 0.45 : 1)
    }
}

File: flutter/library/lib/src/tsp_chip.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

class TspChip extends StatelessWidget {
  const TspChip({super.key, required this.text, this.selected = false, this.disabled = false, this.theme = StarPlanetTheme.sky, this.onTap});
  final String text;
  final bool selected;
  final bool disabled;
  final StarPlanetTheme theme;
  final VoidCallback? onTap;

  @override
  Widget build(BuildContext context) => GestureDetector(
        onTap: disabled ? null : onTap,
        child: Opacity(
          opacity: disabled ? .45 : 1,
          child: Container(
            padding: const EdgeInsets.symmetric(horizontal: 14, vertical: 8),
            decoration: ShapeDecoration(
              color: selected ? theme.selectedFill : theme.surfaceRaised,
              shape: StadiumBorder(side: BorderSide(color: selected ? theme.success : theme.borderDefault)),
            ),
            child: Text(text, style: TextStyle(color: theme.textPrimary, fontWeight: FontWeight.w800)),
          ),
        ),
      );
}
